partner_pricelist_search/tests/test_partner_pricelist_search.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2021 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo.exceptions import UserError
+from odoo.tests import common
+
+
+class TestPartnerPricelistSearch(common.SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.pricelist_1 = cls.env["product.pricelist"].create({
+            "name": "Test pricelist 1",
+        })
+        cls.pricelist_2 = cls.env["product.pricelist"].create({
+            "name": "Test pricelist 2",
+        })
+        cls.customer_1 = cls.env["res.partner"].create({
+            "name": "Test customer 1",
+            "property_product_pricelist": cls.pricelist_1,
+        })
+        cls.customer_2 = cls.env["res.partner"].create({
+            "name": "Test customer 2",
+            "property_product_pricelist": cls.pricelist_2,
+        })
+
+    def test_partner_pricelist_search(self):
+        ResPartner = self.env["res.partner"]
+
+        # Test search '='
+        partners = ResPartner.search([
+            ("property_product_pricelist", "=",
+             self.pricelist_1.id)
+        ])
+        self.assertEqual(partners, self.customer_1)
+
+        # Test search 'in'
+        partners = ResPartner.search([
+            ("property_product_pricelist", "in",
+             (self.pricelist_1 | self.pricelist_2).ids)
+        ])
+        self.assertIn(self.customer_1, partners)
+        self.assertIn(self.customer_2, partners)
+
+        # Test search '!='
+        partners = ResPartner.search([
+            ("property_product_pricelist", "!=",
+             self.pricelist_1.id)
+        ])
+        self.assertNotIn(self.customer_1, partners)
+        self.assertIn(self.customer_2, partners)
+
+        # Test search 'not in'
+        partners = ResPartner.search([
+            ("property_product_pricelist", "not in",
+             (self.pricelist_1 | self.pricelist_2).ids)
+        ])
+        self.assertNotIn(self.customer_1, partners)
+        self.assertNotIn(self.customer_2, partners)
+
+        # Test search not implemented
+        with self.assertRaises(UserError):
+            ResPartner.search([
+                ("property_product_pricelist", "ilike",
+                 "pricelist xx")
+            ])
+
+    def test_show_pricelist_partners(self):
+        res = self.pricelist_1.show_pricelist_partners()
+        self.assertTrue(res["domain"])
```

Comment:
Test the real result
```suggestion
        self.assertEqual(ResPartner.search(res["domain"]), self.customer_1)
```
and the same for the other one

Reply:
I don't want add complexity in the test to avoid future problems if a new módule adds conditions in this domain.
I am satisfied that the method returns an action with domain.

Reply:
Well, this test is executed right after the installation of this module, so it's literally impossible that this fails although you do any extension. The only way this can fail is if you define the tests with the tag post-install.

Reply:
Ok. Changes done.